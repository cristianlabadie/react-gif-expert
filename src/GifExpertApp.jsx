import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (category) => {
    if (categories.includes(category)) return;

    setCategories([category, ...categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={(value) => onAddCategory(value)} />

      <button onClick={onAddCategory}>Agregar categoria</button>

      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};
