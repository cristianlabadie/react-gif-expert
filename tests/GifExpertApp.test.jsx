import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  test("Debe de mostrar el componente correctamente", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de poder escribir en el input y agregar una nueva categoría", () => {
    const category = "Deku";
    render(<GifExpertApp />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: category } });
    fireEvent.submit(form);
    expect(screen.getByText(category).innerHTML).toBeTruthy();
  });

  test("Debe de no agregar una categoría si ya existe", () => {
    const category = "Deku";

    render(<GifExpertApp />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: category } });
    fireEvent.submit(form);
    fireEvent.input(input, { target: { value: category } });
    fireEvent.submit(form);
    expect(screen.getAllByText("Deku").length).toBe(1);
  });
});
