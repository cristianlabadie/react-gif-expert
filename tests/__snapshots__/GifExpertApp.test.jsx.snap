// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
  <button>
    Agregar categoria
  </button>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
